frame/evm/precompile/secp256r1/src/lib.rs

```diff
@@ -0,0 +1,135 @@
+// This file is part of Frontier.
+
+// Copyright (c) Moonsong Labs.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// “secp256r1” is a specific elliptic curve, also known as “P-256” and “prime256v1” curves.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+extern crate alloc;
+
+use p256::ecdsa::{signature::hazmat::PrehashVerifier, Signature, VerifyingKey};
+use fp_evm::{ExitSucceed, PrecompileResult, Precompile, PrecompileHandle, PrecompileOutput};
+
+pub struct P256Verify;
+
+impl P256Verify {
+	/// https://github.com/ethereum/RIPs/blob/master/RIPS/rip-7212.md#precompiled-contract-gas-usage
+	const BASE_GAS: u64 = 3_450;
+}
+
+
+/// Implements RIP-7212 P256VERIFY precompile.
+/// https://github.com/ethereum/RIPs/blob/master/RIPS/rip-7212.md
+impl Precompile for P256Verify {
+
+	fn execute(handle: &mut impl PrecompileHandle) -> PrecompileResult {
+		handle.record_cost(Self::BASE_GAS)?;
+
+		let input = handle.input();
+
+		let result = if verify_impl(input).is_some() {
+			// If the signature verification process succeeds, it returns 1 in 32 bytes format.
+			let mut result = [0u8; 32];
+			result[31] = 1;
+
+			result.to_vec()
+		} else {
+			// If the signature verification process fails, it does not return any output data.
+			[0u8; 0].to_vec()
+		};
+
+		Ok(PrecompileOutput {
+			exit_status: ExitSucceed::Returned,
+			output: result.to_vec(),
+		})
+	}
+}
+
+/// (Signed payload) Hash of the original message
+/// 32 bytes of the signed data hash
+fn message_hash(input: &[u8]) -> &[u8] {
```

Comment:
Please add the inline attribute for theses 3 functions, as there exist only to make the code more readable

Reply:
Fixed on moonbeam side